### DISCUSSION TITLE
Unified variable, function, class, struct declaration

### DISCUSSION THREAD

**Discussion by C-BJ** [carbon-language/carbon-lang#1542] [Language design]
`let main = fn()->i32 {  }`
Functions also have types

`let animal = class{  };`

Such a design can be extended to many aspects

It's just my own strange idea

**Comment by C-BJ** [comment]
Related issue: https://github.com/carbon-language/carbon-lang/issues/1591

**Comment by lexi-nadia** [comment]
Closing this in favor of the issue.
